src/main/java/nl/tudelft/cse1110/andy/result/MetaTestsResult.java

```diff
@@ -1,33 +1,43 @@
 package nl.tudelft.cse1110.andy.result;
 
+import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 
 public class MetaTestsResult {
-    private final boolean wasExecuted;
-    private final int score;
-    private final int totalTests;
-    private final List<MetaTestResult> metaTestResults;
+    private boolean wasExecuted;
+    private int score;
+    private int totalTests;
+    private final ArrayList<MetaTestResult> metaTestResults;
 
-    private MetaTestsResult(boolean wasExecuted, int score, int totalTests, List<MetaTestResult> metaTestResults) {
-        this.wasExecuted = wasExecuted;
-        this.score = score;
-        this.totalTests = totalTests;
-        this.metaTestResults = metaTestResults;
-
-        if(this.metaTestResults.size() > this.totalTests)
-            throw new RuntimeException("Number of meta tests do not match.");
-
-        if(this.score > this.totalTests)
-            throw new RuntimeException("Meta tests score greater than maximum.");
+    private MetaTestsResult() {
+        this.wasExecuted = false;
+        this.score = 0;
+        this.totalTests = 0;
+        this.metaTestResults = new ArrayList<>();
     }
 
     public static MetaTestsResult build(int score, int totalTests, List<MetaTestResult> metaTestResults) {
-        return new MetaTestsResult(true, score, totalTests, metaTestResults);
+        return MetaTestsResult.empty().addResults(score, totalTests, metaTestResults);
     }
 
     public static MetaTestsResult empty() {
-        return new MetaTestsResult(false, 0, 0, Collections.emptyList());
+        return new MetaTestsResult();
+    }
+
+    public MetaTestsResult addResults(int score, int totalTests, List<MetaTestResult> metaTestResults) {
+        if (metaTestResults.size() > totalTests)
+            throw new RuntimeException("Number of meta tests does not match.");
```

Comment:
Maybe add in the exception message something that makes us understand this is a pre-condition being broken?

Reply:
Sure, I will do it in a bit.